### PR TITLE
JITX-3924: Add part categories

### PR DESCRIPTION
**Pull request by ptuckey**
Add parsing support for JITX-specific categories.

### CODE REVIEW

File: utils/parts.stanza [hunk 1]
```diff
@@ -389,6 +391,32 @@ defn string-or-unknown (json: JSON) -> String|UNKNOWN :
...
+    "connector": connector
+    "diode": diode
+    "fuse": fuse
+    "inductor": inductor
+    "microcontroller": microcontroller
+    "oscillator": oscillator
+    "power_supply_chip": power_supply_chip
+    "resistor": resistor
+    "transistor": transistor
+    else: UNKNOWN
```
Comment (PhilippeFerreiraDeSousa) [inline utils/parts.stanza:418]: Hey, here is a compact way to write it, it is less readable but has lower maintenance / less error prone:
```
import lang-utils

val category-values = map(Category, 0 to Category-length)
val string-to-category-value = to-hashtable(to-string, category-values)

defn category-or-unknown (category-str: String) -> Category|UNKNOWN :
  get?(string-to-category-value, category-str, UNKNOWN)
```

For OCDB, preferring readability like you did makes sense, so not a change request

**Review by PhilippeFerreiraDeSousa — APPROVED** [review]

File: utils/parts.stanza [hunk 2]
```diff
@@ -389,6 +391,32 @@ defn string-or-unknown (json: JSON) -> String|UNKNOWN :
...
 
+public defenum Category :
+  capacitor
+  connector
+  diode
+  fuse
+  inductor
+  microcontroller
+  oscillator
+  power_supply_chip
```
Comment (PhilippeFerreiraDeSousa) [inline utils/parts.stanza:402]: Do we want this with `-` instead of `_`?

File: utils/parts.stanza (hunk 1 above)
Comment (ptuckey) [reply utils/parts.stanza:418]: Oh awesome thanks! I was wondering if there was a better way...

I'll leave as is for the reasons you stated.

File: utils/parts.stanza (hunk 2 above)
Comment (ptuckey) [reply utils/parts.stanza:402]: Good call!

The fields within the schema itself all use underscores (to be Python-friendly) - but, here it makes sense to use Stanza idioms.

**Pull request merged**